LICENSE-binary

```diff
@@ -0,0 +1,520 @@
+org.apache.hadoop:hadoop-client
+org.apache.hadoop:hadoop-common
+org.apache.hadoop:hadoop-hdfs
+org.apache.hadoop:hadoop-mapreduce-client-app
+org.apache.hadoop:hadoop-mapreduce-client-common
+org.apache.hadoop:hadoop-mapreduce-client-core
+org.apache.hadoop:hadoop-mapreduce-client-jobclient
+org.apache.hadoop:hadoop-mapreduce-client-shuffle
+org.apache.hadoop:hadoop-yarn-api
+org.apache.hadoop:hadoop-yarn-client
+org.apache.hadoop:hadoop-yarn-common
+org.apache.hadoop:hadoop-yarn-server-common
+org.apache.hadoop:hadoop-yarn-server-web-proxy
+org.apache.httpcomponents:httpclient
+org.apache.httpcomponents:httpcore
+org.apache.orc:orc-core
+org.apache.orc:orc-mapreduce
+org.mortbay.jetty:jetty
+org.mortbay.jetty:jetty-util
+com.jolbox:bonecp
+org.json4s:json4s-ast_2.11
+org.json4s:json4s-core_2.11
+org.json4s:json4s-jackson_2.11
+org.json4s:json4s-scalap_2.11
+com.carrotsearch:hppc
+com.fasterxml.jackson.core:jackson-annotations
+com.fasterxml.jackson.core:jackson-core
+com.fasterxml.jackson.core:jackson-databind
+com.fasterxml.jackson.dataformat:jackson-dataformat-yaml
+com.fasterxml.jackson.module:jackson-module-jaxb-annotations
+com.fasterxml.jackson.module:jackson-module-paranamer
+com.fasterxml.jackson.module:jackson-module-scala_2.11
+com.github.mifmif:generex
+com.google.code.findbugs:jsr305
+com.google.code.gson:gson
+com.google.inject:guice
+com.google.inject.extensions:guice-servlet
+com.twitter:parquet-hadoop-bundle
+commons-beanutils:commons-beanutils-core
+commons-cli:commons-cli
+commons-dbcp:commons-dbcp
+commons-io:commons-io
+commons-lang:commons-lang
+commons-logging:commons-logging
+commons-net:commons-net
+commons-pool:commons-pool
+io.fabric8:zjsonpatch
+javax.inject:javax.inject
+javax.validation:validation-api
+log4j:apache-log4j-extras
+log4j:log4j
+net.sf.jpam:jpam
+org.apache.avro:avro
+org.apache.avro:avro-ipc
+org.apache.avro:avro-mapred
+org.apache.commons:commons-compress
+org.apache.commons:commons-math3
+org.apache.curator:curator-client
+org.apache.curator:curator-framework
+org.apache.curator:curator-recipes
+org.apache.directory.api:api-asn1-api
+org.apache.directory.api:api-util
+org.apache.directory.server:apacheds-i18n
+org.apache.directory.server:apacheds-kerberos-codec
+org.apache.htrace:htrace-core
+org.apache.ivy:ivy
+org.apache.mesos:mesos
+org.apache.parquet:parquet-column
+org.apache.parquet:parquet-common
+org.apache.parquet:parquet-encoding
+org.apache.parquet:parquet-format
+org.apache.parquet:parquet-hadoop
+org.apache.parquet:parquet-jackson
+org.apache.thrift:libfb303
+org.apache.thrift:libthrift
+org.codehaus.jackson:jackson-core-asl
+org.codehaus.jackson:jackson-mapper-asl
+org.datanucleus:datanucleus-api-jdo
+org.datanucleus:datanucleus-core
+org.datanucleus:datanucleus-rdbms
+org.lz4:lz4-java
+org.spark-project.hive:hive-beeline
+org.spark-project.hive:hive-cli
+org.spark-project.hive:hive-exec
+org.spark-project.hive:hive-jdbc
+org.spark-project.hive:hive-metastore
```

Comment:
Hi @srowen , Do we need to add these libs for current master branch?
`org.apache.hive:hive-beeline`
`org.apache.hive:hive-cli`
`org.apache.hive:hive-exec`
`org.apache.hive:hive-jdbc `
`org.apache.hive:hive-metastore`

Reply:
These are listed in the current license file in master. If they are it's because they showed up in the transitive dependencies in compile/runtime scope. Are you asking if they can be removed? 

Heh, I've lost track, do we use this Hive fork anymore now? if not, they can be removed.

You can look at `mvn dependency:tree` to see if they still appear, if in doubt.

Reply:
I mean these jars:
```
[root@spark-3267648 apache-spark]# mvn dependency:tree -Phadoop-3.2 | grep "org.apache.hive" | sort | uniq
[INFO] +- org.apache.hive.hcatalog:hive-hcatalog-core:jar:2.3.6:test
[INFO] +- org.apache.hive:hive-common:jar:2.3.6:compile
[INFO] |  +- org.apache.hive:hive-common:jar:2.3.6:provided
[INFO] +- org.apache.hive:hive-contrib:jar:2.3.6:test
[INFO] +- org.apache.hive:hive-exec:jar:core:2.3.6:compile
[INFO] |  +- org.apache.hive:hive-exec:jar:core:2.3.6:provided
[INFO] +- org.apache.hive:hive-llap-client:jar:2.3.6:test
[INFO] +- org.apache.hive:hive-llap-common:jar:2.3.6:compile
[INFO] |  \- org.apache.hive:hive-llap-common:jar:2.3.6:provided
[INFO] +- org.apache.hive:hive-metastore:jar:2.3.6:compile
[INFO] |  +- org.apache.hive:hive-metastore:jar:2.3.6:provided
[INFO] +- org.apache.hive:hive-serde:jar:2.3.6:compile
[INFO] |  +- org.apache.hive:hive-serde:jar:2.3.6:provided
[INFO] |  +- org.apache.hive:hive-service-rpc:jar:2.3.6:compile
[INFO] |  |  \- org.apache.hive:hive-service-rpc:jar:2.3.6:provided
[INFO] +- org.apache.hive:hive-shims:jar:2.3.6:compile
[INFO] |  +- org.apache.hive:hive-shims:jar:2.3.6:provided
[INFO] \- org.apache.hive:hive-storage-api:jar:2.6.0:compile
[INFO] |  +- org.apache.hive:hive-vector-code-gen:jar:2.3.6:compile
[INFO] |  |  +- org.apache.hive:hive-vector-code-gen:jar:2.3.6:provided
[INFO] |  |  +- org.apache.hive.shims:hive-shims-0.23:jar:2.3.6:provided
[INFO] |  +- org.apache.hive.shims:hive-shims-0.23:jar:2.3.6:runtime
[INFO] |  +- org.apache.hive.shims:hive-shims-common:jar:2.3.6:compile
[INFO] |  |  +- org.apache.hive.shims:hive-shims-common:jar:2.3.6:provided
[INFO] |  |  \- org.apache.hive.shims:hive-shims-scheduler:jar:2.3.6:provided
[INFO] |  \- org.apache.hive.shims:hive-shims-scheduler:jar:2.3.6:runtime
```

Reply:
We might need to list anything that's in `compile` scope. Really the question is whether these end up in any artifact that we distribute. Really these would appear in the binary distribution of Spark if anywhere. If we are going to distribute a binary release based on Hadoop 3.2 / Hive 2.3, and I'm guessing we must, then yeah these `compile` scope deps should be listed, and technically we need to add their `NOTICE` contents to our `NOTICE` file. I can talk to you about ways to do that if necessary.

As a sanity check, do all of those `compile` scope dependencies look like they are directly needed by Spark?

Reply:
Thank you @srowen